src/frontend/src/expr/type_inference.rs

```diff
@@ -255,10 +255,25 @@ fn build_type_derive_map() -> HashMap<FuncSign, DataTypeName> {
         FuncSign::new(E::RoundDigit, vec![T::Decimal, T::Int32]),
         T::Decimal,
     );
+    // bitwise operator
+    build_binary_atm_funcs(
+        &mut map,
+        &[
+            E::PgBitwiseShiftLeft,
+            E::PgBitwiseShiftRight,
+            E::BitwiseAnd,
+            E::BitwiseOr,
+            E::BitwiseNot,
+            E::BitwiseXor,
```

Comment:
* `BitwiseNot` is not binary.
* See comment above for different typing rule of shifting.